marshaler.go

```diff
@@ -108,24 +110,36 @@ func NewEncoder(w io.Writer) *Encoder {
 //   a quoted multi-line TOML string.
 func (enc *Encoder) Encode(v interface{}) error {
 	var b []byte
+
 	var ctx encoderCtx
+
 	b, err := enc.encode(b, ctx, reflect.ValueOf(v))
 	if err != nil {
-		return err
+		return fmt.Errorf("Encode: %w", err)
 	}
+
 	_, err = enc.w.Write(b)
-	return err
+	if err != nil {
+		return fmt.Errorf("Encode: %w", err)
+	}
+
+	return nil
 }
 
+var errUnsupportedValue = errors.New("unsupported encode value kind")
+
+//nolint:cyclop
 func (enc *Encoder) encode(b []byte, ctx encoderCtx, v reflect.Value) ([]byte, error) {
-	switch i := v.Interface().(type) {
-	case time.Time: // TODO: add TextMarshaler
+	i, ok := v.Interface().(time.Time)
+	//nolint:godox
```

Comment:
we can leave godox and the nolint, so when the TODOs will be removed, the nolint will be flagged as well by nolintlint.
up to you, if we do that, we don't risk forgetting to reactivate it but it's annoying to add the nolint, agree.

Reply:
I've been going back and forth on this. I use TODOs because I'm used to them when I work on a code base. But as contributors, would you think it would be beneficial if I took the time to create a github issue for those instead?

Reply:
I think it depends on how you use TODOs.  If you use TODOs to tag bugs, then having godox flag them is good. If you use TODOs to mark non-bug cleanups or future enhancements, then godox feels really annoying.

Perhaps there's a middle ground: tag bugs with BUG or FIXME comments, and use TODOs for non-bug stuff.  We can tell godox to ignore TODO comments.

Also, I'm sure it would be helpful if the TODO/BUG/FIXME comments referenced a GH issue.

Reply:
I'd say godox ensures you ll review all your TODOs before it's merged, and add a nolint in case you don't want to do it right now.
The cost of adding //nolint:godox is quite small, even though it's ugly, I'll completely admit it.
Not stating any pref here, letting @pelletier decide, just trying to see the other side of the coin.

Reply:
I've been thinking about it: I think we should keep godox, and I'll be better about keeping track of things in the issues tracker. Reasoning is: I use TODOs to mark future cleanups, potential enhancements that could be made, and as a rare case mark incomplete features when I'm working on my own. In the spirit of using the linter to make it easier for people to contribute to the project, I think having the linter flag TODOs would help increase the discoverability of tasks (I'm assuming it's easier for people to see potential work in issues rather than in the code), and would help catch missing TODOs from external pull requests.

Feel free to add `//nolint:godox` at the moment. I will do a pass on the existing TODOs and create issues for them.